api/submission/index.ts

```diff
@@ -1,5 +1,66 @@
 import axios from 'axios'
 import { baseURL, getAuthedHeaders } from '../common'
+import { User } from '../models/user'
 
-export const getMySubmissions = () =>
-  axios.get(`${baseURL}/my-submissions`, { headers: getAuthedHeaders() }).then(({ data }) => data)
\ No newline at end of file
+interface Submission {
+  id: string
+  status: 'pending' | 'in progress' | 'done'
+  title: string
+  symptoms: string
+  created_at: string
+  prescription: string
+  doctor?: User
+  patient: User
+}
+type BadStatus = 'pending' | 'in progress' | 'in_progress' | 'done'
+type BadSubmission = Omit<Submission, 'status'> & { status: BadStatus }
+function cleanSubmission(s: BadSubmission) {
+  if (s.status === 'in_progress') {
+    return { ...s, status: 'in progress' }
+  }
+  return s
```

Comment:
now that I think about it, HERE we should do `return s as Submission` and then you don't need to do the castings we do at line 29 and line 37